tests/data_processing/test_get_flight_stats.py

```diff
@@ -0,0 +1,139 @@
+import json
+import unittest
+
+import pandas as pd
+import pandas.testing as pd_testing
+
+from src.data.get_flight_stats import FlightStats
+from src.utils.json_converter import DataConverter
+
+
+class TestFlightStats(unittest.TestCase):
+    FLIGHT_COUNT = [
+        'total_arrivals',
+        'total_departures'
+    ]
+
+    def assertDataframeEqual(self, a: pd.DataFrame, b: pd.DataFrame, msg: None = None) -> None:
+        try:
+            pd_testing.assert_frame_equal(a, b)
+        except AssertionError as e:
+            raise self.failureException(msg) from e
+
+    def setUp(self) -> None:
+        self.addTypeEqualityFunc(pd.DataFrame, self.assertDataframeEqual)
+
+    def test_count_flights_structure(self) -> None:
+        self.data_path = 'data/tests/model_test_subset_dataset.json'
+        with open(self.data_path, 'r', encoding='utf-8') as f:
+            self.raw_data = json.load(f)
+        self.airport_data = DataConverter(self.raw_data)
+        airport_df = self.airport_data.airports_df
+        flights_df = self.airport_data.flights
+        flights_df = pd.concat([flights_df, airport_df['icao']], axis=1, join='outer')
+
+        flights_stats = FlightStats(flights_df)
+        # check output dataframe types are int
+        self.assertTrue(all(isinstance(count, int) for count in flights_stats.flight_stats_df['total_arrivals']))
+        self.assertTrue(all(isinstance(count, int) for count in flights_stats.flight_stats_df['total_departures']))
+        # check output shape
+        self.assertEqual(flights_stats.flight_stats_df.shape[1], len(self.FLIGHT_COUNT))
+        # check column names are as expected
+        self.assertEqual(set(flights_stats.flight_stats_df.columns), set(self.FLIGHT_COUNT))
+        # check output is a dataframe
+        self.assertIsInstance(flights_stats.flight_stats_df, pd.DataFrame)
+
+    def test_count_flights_logic_base_case(self) -> None:
+        # base test case
+        fake_input_flights_df = pd.DataFrame(
+            {
+                'icao': 'LBG',
+                'flights': [
+                    [
+                        {'origin_iata': 'LBG'},
+                        {'origin_iata': 'LBG'},
+                        {'origin_iata': 'LBG'},
+                        {'origin_iata': 'SAY'},
+                        {'origin_iata': 'YVR'}
+                    ]
+                ]
+            }
+        )
+        flights_stats = FlightStats(fake_input_flights_df)
+
+        fake_output_flights_df = pd.DataFrame(
+            {
+            'total_arrivals': [2],
+            'total_departures': [3]
+            }
```

Comment:
Is this correct? If `origin_iata` is somewhere else then is it correct to assume the destination must always be `LBG`?

Reply:
Yeah I was counting the arrivals/departures based on the origin_iata matching or not matching the airport iata - is that an unsafe assumption?

Reply:
I think it's fine, I was just checking